src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -68,4 +71,34 @@ public CompletableFuture<SecurityAnalysisReport> run(String workingVariantId, Se
 
     abstract SecurityAnalysisReport runSync(String workingVariantId, SecurityAnalysisParameters securityAnalysisParameters, ContingenciesProvider contingenciesProvider,
                                             ComputationManager computationManager, List<OperatorStrategy> operatorStrategies, List<Action> actions);
+
+    public static PostContingencyComputationStatus postContingencyStatusFromNRStatus(NewtonRaphsonStatus status) {
+        switch (status) {
+            case CONVERGED:
+                return PostContingencyComputationStatus.CONVERGED;
+            case MAX_ITERATION_REACHED:
+                return PostContingencyComputationStatus.MAX_ITERATION_REACHED;
+            case SOLVER_FAILED:
+                return PostContingencyComputationStatus.SOLVER_FAILED;
+            case NO_CALCULATION:
+                return PostContingencyComputationStatus.NO_IMPACT;
+            case UNREALISTIC_STATE:
+                return PostContingencyComputationStatus.FAILED;
+            default:
+                return PostContingencyComputationStatus.CONVERGED;
+        }
+    }
+
+    public static LoadFlowResult.ComponentResult.Status loadFlowResultStatusFromNRStatus(NewtonRaphsonStatus status) {
+        switch (status) {
+            case CONVERGED:
+                return LoadFlowResult.ComponentResult.Status.CONVERGED;
+            case MAX_ITERATION_REACHED:
+                return LoadFlowResult.ComponentResult.Status.MAX_ITERATION_REACHED;
+            case SOLVER_FAILED:
+                return LoadFlowResult.ComponentResult.Status.SOLVER_FAILED;
+            default:
+                return LoadFlowResult.ComponentResult.Status.FAILED;
```

Comment:
we should explicitely map all the enum and throws an exception in the defaut case.